tests/integration/test_core.py

```diff
@@ -53,6 +55,33 @@ async def test_hello_kubecon_cache_header(ingress_ip: str):
     assert "content-cache-k8s" in response.headers["X-Cache-Status"]
 
 
+@pytest.mark.asyncio
+@pytest.mark.abort_on_fail
+async def test_service_reachable(service_ip: str):
+    """
+    arrange: given charm has been built, deployed and related to a dependent application
+    act: when the dependent application is queried via the service
+    assert: then the response is HTTP 200 OK.
+    """
+    response = requests.get(f"http://{service_ip}", timeout=5)
+
+    assert response.status_code == 200
+
+
+async def test_report_visits_by_ip(app: Application):
```

Comment:
You'll need to unit test this method by mocking the action execution instead to make sure you are getting the IPs as expected for a given file content. You should test correct identification of the lines, counting and ordering (if needed).

Reply:
The unit tests are already written for this method, I wasn't mocking the action so now that will be changed on the next commit

Reply:
Nvm then